cpp/src/rolling/grouped_rolling.cu

```diff
@@ -371,26 +373,27 @@ std::unique_ptr<column> time_range_window_ASC(column_view const& input,
 /// Each group in the input timestamp column must be sorted,
 /// with null values clustered at either the start or the end of each group.
 /// If there are no nulls for any given group, (nulls_begin, nulls_end) == (0,0).
-std::tuple<rmm::device_vector<size_type>, rmm::device_vector<size_type>>
+std::tuple<rmm::device_uvector<size_type>, rmm::device_uvector<size_type>>
 get_null_bounds_for_timestamp_column(column_view const& timestamp_column,
-                                     rmm::device_uvector<size_type> const& group_offsets)
+                                     cudf::device_span<size_type const> group_offsets,
+                                     rmm::cuda_stream_view stream)
 {
   // For each group, the null values are themselves clustered
   // at the beginning or the end of the group.
   // These nulls cannot participate, except in their own window.
 
   // If the input has n groups, group_offsets will have n+1 values.
   // null_start and null_end should eventually have 1 entry per group.
-  auto null_start = rmm::device_vector<size_type>(group_offsets.begin(), group_offsets.end() - 1);
-  auto null_end   = rmm::device_vector<size_type>(group_offsets.begin(), group_offsets.end() - 1);
+  auto null_start = cudf::detail::make_device_uvector_async(group_offsets, stream);
+  auto null_end   = cudf::detail::make_device_uvector_async(group_offsets, stream);
 
   if (timestamp_column.has_nulls()) {
-    auto p_timestamps_device_view = column_device_view::create(timestamp_column);
-    auto num_groups               = group_offsets.size() - 1;
+    auto p_timestamps_device_view = column_device_view::create(timestamp_column, stream);
+    auto num_groups               = group_offsets.size();
 
     // Null timestamps exist. Find null bounds, per group.
     thrust::for_each(
-      thrust::device,
+      rmm::exec_policy(stream),
```

Comment:
I think after looking at this algorithm quickly that it will update each value in `null_start` and `null_end`. If that is correct, we could allocate space instead of copying from `group_offsets`.

Reply:
Is it up to uvector conversion PRs to improve algorithms?

Reply:
No it is not. I will make a follow-on PR that improves the algorithm to only copy memory when nulls don't exist.

Reply:
BTW, sorry I misunderstood your original comment. This kind of change is definitely intimately related to converting to device_uvector, since it is about initialization.